### ISSUE TITLE
[Feature Request]: teal::init to return shinyApp instead of a list

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal#915]
`teal::init` to return `"teal_app"` object (simple S3 class). For this class we can have a print method which can look as follows:

```
print.teal_app <- function(x, ...) {
  cat("teal app <more information about the app>")
  shiny::shinyApp(x$ui, x$srv) 
}

```
Thanks to above "teal_app" object will preserve previous features (list of ui, server, ui_args, ... etc) and could be run without shinyApp(x$ui, x$server) . This would shorten the apps code:

```
# from
app <- teal::init(...)
shinyApp(app$ui, app$server)

# to
teal::init(...)
```

With the S3 class first and the second in above code would be possible.

Alternatively, print.teal_app could just print information about the app with note use shinyApp(x$ui, x$server) to run the app.

Please note that shiny works in very similar way
https://github.com/rstudio/shiny/blob/370ba1f288d3f759456ef3ef6e768ee1d7fda677/R/shinyapp.R#L552

**Comment by chlebowa** [comment]
Do you mean just adding `shinyApp(ui = res$ui, server = res$server)` as the return value of `init`?

> Nothing will be broken due to this, 

Besides the fact that you can't do `shinyApp(app$ui, app$server)` anymore, do I have this correct?

**Comment by gogonzo** [comment]
@chlebowa @pawelru true, I have not noticed `shinyApp` doesn't return a list. Forget about this.

----

Initially I thought that `teal::init` can return object of class `teal.app` and we can have a `print.teal.app` which runs the app. Since it is not so easy I don't push for this and I suggest to refine this first ;)

**Comment by vedhav** [comment]
I don't think we should return `shiny.appobj` from `teal::init()` if we did this we would have to support other arguments to be passed into `init` like `enableBookmarking`.
I misinterpreted the feature request when I first read it. I just thought that we're replacing the `runApp(app)` into `shinyApp(app$ui, app$server)` as the former is meant for interactive calling of shiny app and has known problems when it comes to deployment.

**Comment by gogonzo** [comment]
Idea doesn't make sense. Let's pay attention to https://github.com/insightsengineering/teal/issues/1448
